test/testdata/resolver/fuzz_multiple_sigs.rb

```diff
@@ -2,10 +2,10 @@
 # fuzzed test from https://github.com/sorbet/sorbet/issues/1166
 def r
   sig{} # error: Malformed `sig`: No return type specified. Specify one with .returns()
-# ^^^^^ error: Method `sig` does not exist on `Object`
```

Comment:
ooc: why has this changed?

Reply:
This has changed because the same code that flattens methods will pull sigs unconditionally out as well. This means that
```ruby
def r
  sig {}
  def f; end
end
```
becomes equivalent to
```ruby
def r;
sig{}
def f; end
```
which makes us treat it as a top-level definition, which changes the error message as well.